### PR TITLE
Support -it vddk for virt-v2v

### PR DESCRIPTION
**Pull request by xiaodwan**
1) Handle interactive operation with VMware Vcenter to lauch 'Bash'
2) Autoset vddk_libdir and vddk_thumbprint if not set by user
3) Use user's value of vddk_libdir and vddk_thumbprint if set
4) Autoclean mount records after v2v command finishes

Signed-off-by: Xiaodai Wang <xiaodwan@redhat.com>

### CODE REVIEW

**Comment by xiaodwan** [comment]
(1/2) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.esx.vm.5_5.linux.latest6.arch_x86_64.raw_f.ISCSI.rhv_upload.rhv_direct.rhv_verifypeer.preallocated.it_vddk: PASS (1140.70
s)
 (2/2) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.esx.vm.5_5.linux.latest6.arch_x86_64.raw_f.ISCSI.rhv_upload.rhv_direct.rhv_verifypeer.preallocated.it_default: PASS (2828.05 s)

**Comment by xiaodwan** [comment]
@chloerh @chunfuwen please help to review this pr, thanks.

**Comment by xiaodwan** [comment]
(1/2) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.esx.vm.6_7.linux.latest6.arch_i386.raw_f.NFS.rhv_upload.rhv_direct.rhv_verifypeer.nopreallocated.it_vddk: PASS (959.51 s)
 (2/2) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.esx.vm.6_7.linux.latest6.arch_i386.raw_f.NFS.rhv_upload.rhv_direct.rhv_verifypeer.nopreallocated.it_default: PASS (1978.54 s)

File: virttest/remote.py [hunk 1]
```diff
@@ -156,7 +158,7 @@ def handle_prompts(session, username, password, prompt, timeout=10,
                  r"[Cc]onnection.*closed", r"[Cc]onnection.*refused",
                  r"[Pp]lease wait", r"[Ww]arning", r"[Ee]nter.*username",
                  r"[Ee]nter.*password", r"[Cc]onnection timed out", prompt,
-                 r"Escape character is.*"],
+                 r"Escape character is.*", r"Command>"],
```
Comment (chunfuwen) [inline virttest/remote.py]: separate r"<Command>" into new line

File: virttest/remote.py [hunk 2]
```diff
@@ -211,6 +213,20 @@ def handle_prompts(session, username, password, prompt, timeout=10,
             elif match == 13:  # console prompt
                 logging.debug("Got console prompt, send return to show login")
                 session.sendline()
+            elif match == 14:  # VMware vCenter command prompt
+                logging.debug(
+                    "Got VMware VCenter prompt, send 'shell' to launch bash")
+                if not shell_enabled:
```
Comment (chunfuwen) [inline virttest/remote.py]: shell_enabled is hard to 'false' by you, so whatever the execution will enter here. Maybe just remove the judgement,and just run 'shell.set --enabled True ' and 'shell' squencially.

File: virttest/utils_v2v.py [hunk 3]
```diff
@@ -110,6 +123,54 @@ def get_cmd_options(self, params):
         self.new_name = self.params.get('new_name')
         self.net_vm_opts = ""
 
+        def _get_input_transport_options():
```
Comment (chunfuwen) [inline virttest/utils_v2v.py]: get_input_transport_options change to compose_input_transport_xx ?

File: virttest/utils_v2v.py [hunk 4]
```diff
@@ -110,6 +123,54 @@ def get_cmd_options(self, params):
...
+            """
+            Set input transport options for v2v
+            """
+            options = ''
+            if self.input_transport is None:
```
Comment (chunfuwen) [inline virttest/utils_v2v.py]: those code doesn't need if you just return options at the last line of this methold

File: virttest/utils_v2v.py [hunk 5]
```diff
@@ -869,3 +935,26 @@ def cleanup_constant_files(params):
                 params.get("vpx_passwd_file")]
 
     map(os.remove, [x for x in tmpfiles if x and os.path.isfile(x)])
+
+
+def get_vddk_thumbprint(host, password, prompt=r"[\#\$]"):
+    """
+    Get vddk thumbprint from VMware vCenter
+
+    :param host: Hostname or IP address
```
Comment (chunfuwen) [inline virttest/utils_v2v.py]: Hostname --> hostname

File: virttest/utils_v2v.py [hunk 6]
```diff
@@ -869,3 +935,26 @@ def cleanup_constant_files(params):
...
+    :param password: Password
+    :param prompt: Shell prompt (regular expression)
+    """
+    cmd = 'openssl x509 -in /etc/vmware-vpx/ssl/rui.crt -fingerprint -sha1 -noout'
+
+    try:
```
Comment (chunfuwen) [inline virttest/utils_v2v.py]: no need try ... except since you just raise exception when happen

File: virttest/remote.py (hunk 1 above)
Comment (xiaodwan) [reply virttest/remote.py]: Done

File: virttest/remote.py (hunk 2 above)
Comment (xiaodwan) [reply virttest/remote.py]: Done

File: virttest/utils_v2v.py (hunk 3 above)
Comment (xiaodwan) [reply virttest/utils_v2v.py]: Done

File: virttest/utils_v2v.py (hunk 5 above)
Comment (xiaodwan) [reply virttest/utils_v2v.py]: Done

File: virttest/utils_v2v.py (hunk 6 above)
Comment (xiaodwan) [reply virttest/utils_v2v.py]: Done

File: virttest/utils_v2v.py (hunk 4 above)
Comment (xiaodwan) [reply virttest/utils_v2v.py]: We need it to return if user didn't pass any input_transport options.

**Pull request merged**